### PR TITLE
Get num hops

### PR DESCRIPTION
**Pull request by Kheims**
## Proposed changes

Implementation of https://github.com/TristanBilot/mlx-graphs/issues/46.
I have used misc as the module name, do you have a better name in mind or maybe suggest to relocate it in other existing modules ? I agree it is related to the models should we have a utils/ or model_info/ or something like that located in nn ?

## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

File: mlx_graphs/utils/misc.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import mlx.nn as nn
+
+
+def get_num_hops(model: nn.Module) -> int:
+    """
+    Returns the number of hops the model is aggregating information
+    from.
```
Comment (francescofarina) [inline mlx_graphs/utils/misc.py:7]: Specify that this works only for networks based on `MessagePassing`

File: docs/source/api/utils/misc.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+Misc
+=======
+
+
+.. currentmodule:: mlx_graphs.utils.misc
+
+
+.. autosummary::
+	:toctree: _autosummary_sorting
```
Comment (francescofarina) [inline docs/source/api/utils/misc.rst:12]: `_autosummary_misc`

**Review by francescofarina — CHANGES_REQUESTED** [review]
Let's keep it in `mlx_graphs/utils` but let's find a better name than `misc`. Perhaps it should stay in `topology.py`?

File: mlx_graphs/utils/misc.py
Comment (francescofarina) [inline mlx_graphs/utils/misc.py:1]: rm file?

**Comment by Kheims** [comment]
> Let's keep it in `mlx_graphs/utils` but let's find a better name than `misc`. Perhaps it should stay in `topology.py`?

Although topology seems graph's structure related, I think it's indeed better to put it there.

**Review by francescofarina — APPROVED** [review]
Looks good - please remove `misc.py` before merging

File: mlx_graphs/utils/misc.py
Comment (Kheims) [reply mlx_graphs/utils/misc.py:1]: Yes just did

File: mlx_graphs/utils/topology.py [hunk 3]
```diff
@@ -62,3 +63,38 @@ def is_directed(edge_index: mx.array, edge_features: Optional[mx.array] = None)
...
+        ...         self.conv1 = GCNConv(4, 16)
+        ...         self.conv2 = GCNConv(16, 16)
+        ...         self.lin = nn.linear(16, 2)
+        ...
+        ...     def __call__(self, edge_index: mx.array, node_features: mx.array) -> mx.array:
+        ...         x = nn.relu(self.conv1(node_features, edge_index))
+        ...         x = self.conv2(node_features, edge_index)
+        ...         return self.lin(x)
+        >>> get_num_hops(GNN())
+        2
```
Comment (francescofarina) [inline mlx_graphs/utils/topology.py:92]: @Kheims can you please make this a code-block?

**Pull request merged**